module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,65 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <array>
+#include <iostream>
+#include <utility>
+#include <vector>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedData;
+    compressedData.reserve(width * height);
+
+    for (const auto& row : bitmap) {
+        if (row.empty())
+            continue;
+
+        uint8_t prevElem = row.front();
+        uint8_t cnt = 1;
+
+        for (auto it = row.begin(); it != row.end(); it++) {
+            if (*it == prevElem && it != row.begin()) {
+                cnt++;
+            }
+            if (*it != prevElem || it == std::prev(row.end())) {
+                compressedData.emplace_back(std::make_pair(prevElem, cnt));
+                cnt = 1;
+            }
+            prevElem = *it;
+        }
+    }
```

Comment:
Nice code :)
you can also use while loop when you work with iterators. 
Look at my solution:

```C++
  for (const auto& row : map) {
    auto current = row.begin();
    while (current != row.end()) {
      uint8_t color = *current;
      auto index = std::find_if_not(current, end(row),
        [color](uint8_t other){
            return color == other;
        });
      auto occurrence = std::distance(current, index);
      bitmap.push_back({color, occurrence});
      current = index;
    }
  }
```

Reply:
Thanks for sharing your alternate solution